include/swaybar/config.h

```diff
@@ -65,6 +76,16 @@ uint32_t parse_position(const char *position);
 char *parse_font(const char *font);
 
 /**
+ * Parse display mode dock|hide|invisible.
+ */
+enum display_mode_types parse_display_mode(const char *display_mode);
+
+/**
+ * Parse hidden state show|hide.
+ */
+enum hidden_states parse_hidden_state(const char *hidden_state);
+
```

Comment:
Regarding macros, I don't really want to justify my approach here. Take it on faith that I know what I'm doing.

>Using the correct type of a variable also works in all cases and is certainly not an anti-pattern.

Well, if the variable is an anonymous struct, then you either have to redefine it in the sizeof argument (i.e. `sizeof(arr) / sizeof(struct { ... })`) or use `sizeof(arr) / sizeof(arr[0])`. You can't use `sizeof(array) / (sizeof(void *) + sizeof(void *))`, for example, if the struct is just two pointers, because of alignment issues. So `sizeof(arr) / sizeof(arr[0])` is the most general case.